src/Controls/src/Core/Handlers/Items/iOS/ObservableGroupedSource.cs

```diff
@@ -143,7 +143,14 @@ void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 				return;
 			}
 
-			_collectionView.BeginInvokeOnMainThread(() => CollectionChanged(args));
+			if (!ApplicationModel.MainThread.IsMainThread)
+			{
+				ApplicationModel.MainThread.BeginInvokeOnMainThread(() => CollectionChanged(args));
+			}
+			else
+			{
+				CollectionChanged(args);
+			}
```

Comment:
In general is a good idea to try to not use negations in an if when there is an esle clause: 

> Robert Martin's Clean Code, page 302: Negatives are just a bit harder to understand than positives. So, when possible, conditionals should be expressed as positives.

Would I block a PR from landing because of that, must certainly not and will let the author decide, but reading 'not IsMainThread' is harder and easier to miss specially when it is '!IsMainThread', I have made the error of typing '|IsMainThread' in a piece of code in which the | was valid.